README.md

```diff
@@ -1,58 +1,103 @@
-
-Controllers utilize XDS protocol to discover exported services in federated meshes.
-Controllers are deployed with sidecars, so cross-cluster connections between controllers are secured with Istio mTLS.
-
-## Motivation
-
-In this deployment model, independent meshes deployed in different clusters can connect services without configuring
-access to the k8s api-server in remote clusters. This allows to achieve multi-cluster connectivity for meshes managed
-by different teams in different clusters.
-
-## Development
-
-### Prerequisites
-1. Go 1.22+
-2. protoc 3.19.0+
-3. protoc-gen-go v1.30.0+
-4. protoc-get-golang-deepcopy
-
-### Commands
-
-1. Compile controller:
-    ```shell
-    make
-    ```
-1. Run unit tests:
-    ```shell
-    make test
-    ```
-1. Build image:
-    ```shell
-    HUB=quay.io/maistra-dev TAG=test make docker-build
-    ```
-1. Push image:
-    ```shell
-    HUB=quay.io/maistra-dev TAG=test make docker-push
-    ```
-1. Run e2e tests:
-    ```shell
-    make e2e
-    ```
-1. Run e2e tests with specific Istio version and custom controller image:
-    ```shell
-    HUB=quay.io/maistra-dev TAG=test ISTIO_VERSION=1.23.0 make e2e
-    ```
-1. Run specific test suites:
-    ```shell
-    TEST_SUITES="spire" make e2e
-    ```
-1. Customize federation controller image used in tests (`TAG` is ignored if `USE_LOCAL_IMAGE=true` or not set):
-   ```shell
-   USE_LOCAL_IMAGE=false HUB=quay.io/maistra-dev TAG=0.1 make e2e
-   ```
+This project implements Istio mesh federation using a Kubernetes controller that provides an API
+for managing multi-mesh communication, implements service discovery and automates the management of Istio configuration.
+
+Mesh federation enables mTLS communication between applications across mesh boundaries.
+Each mesh can federate a subset of its services to allow applications from other meshes to connect to these services.
+
+Federated services are exposed on a passthrough gateway, so mTLS is not terminated at the edge of the cluster,
+and authorization can be performed by the federated application.
+
+## Multi-primary vs federation
+
+[Multi-primary](https://istio.io/latest/docs/setup/install/multicluster/multi-primary_multi-network/) and
+[primary-remote](https://istio.io/latest/docs/setup/install/multicluster/primary-remote_multi-network/) topologies
+are great solutions for expanding single mesh to multiple k8s clusters for better system resiliency and higher availability.
+However, these solutions do not fit well in the following cases:
+
+1. Decentralized control and ownership of clusters.
+
+    **Use case**: Different teams or departments manage their own clusters and control planes independently.
+
+    **Reason**: Federation allows each team to maintain autonomy over their cluster’s Istio configurations while still enabling
+    selective cross-mesh communication.
+
+1. Simplified networking between clusters.
```

Comment:
I think it's more about smaller attack surface right? Because we don't need to share API server credentials and don't need to publicly expose the API server to make it work

Reply:
Not quite. Direct access to kube-apiserver is not insecure per se, because users can enable various security mechanisms, like VPN, firewall, TLS, authn, authz, RBAC etc. But in multi-mesh deployments, environments are independent from each other, and enabling all those mechanisms may not be possible. Suggesting that exposing kube-apiserver is insecure in general, might suggest that single mesh multi-cluster solutions are insecure by design, while it's true.